### ISSUE TITLE
bdevperf w/iscsi initiator to SPDK iscs_tgt; unexpected errors w/reset flag and high queue depth 

### ISSUE THREAD

**Issue by peluse** [spdk/spdk#454]
Please use the issue tracker only for reporting suspected issues.

See [The SPDK Community Page](http://www.spdk.io/community/) for other SPDK communications channels.

Discovered while debugging an issue with crypto and this same config; resolved the crypto issue and noticed this still happened even w/only nvme back end or malloc.  Steps here are w/nvme back end. Note that in my setup I use two different machines for init & tgt.

## Expected Behavior
INITIATOR:
[peluse@pels-64 bdevperf]$ sudo ./bdevperf -c ./bdev.conf -q 64 -o 4096 -t 30 -w reset
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x1 -m 1024 --legacy-mem --file-prefix=spdk_pid9529 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid9529/mp_socket
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
BDEVPERF SEND RESET
BDEVPERF SEND RESET
 Logical core: 0
 iSCSI0              :   18721.87 IO/s      73.13 MB/s
 =====================================================
 Total               :   18721.87 IO/s      73.13 MB/s

TARGET
Starting program: /home/peluse/spdk/app/iscsi_tgt/iscsi_tgt -c i.conf
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.26-30.fc27.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: iscsi --no-shconf -c 0x1 --legacy-mem --file-prefix=spdk_pid347043 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid347043/mp_socket
[New Thread 0x7ffff23ff700 (LWP 347047)]
[New Thread 0x7ffff1bfe700 (LWP 347048)]
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
[New Thread 0x7ffbf11ff700 (LWP 347049)]
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
EAL: PCI device 0000:06:00.0 on NUMA socket 0
EAL:   probe driver: 8086:953 spdk_nvme
conn.c: 341:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (143.182.136.79) on iqn.2016-06.io.spdk:disk1 tgt_node1 (143.182.137.32:3260,1), ISID=80732eea0000, TSIH=1, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 781:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 781:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
conn.c: 461:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 1): 0
## Current Behavior
When cranking the OIO up to just 128:
**INITIATOR:**
[peluse@pels-64 bdevperf]$ sudo ./bdevperf -c ./bdev.conf -q 128 -o 4096 -t 30 -w reset
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x1 -m 1024 --legacy-mem --file-prefix=spdk_pid9554 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid9554/mp_socket
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
BDEVPERF SEND RESET
**< HANGS HERE>**

**TARGET: (note the unexpected errors)**
Starting program: /home/peluse/spdk/app/iscsi_tgt/iscsi_tgt -c i.conf
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.26-30.fc27.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: iscsi --no-shconf -c 0x1 --legacy-mem --file-prefix=spdk_pid347099 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid347099/mp_socket
[New Thread 0x7ffff23ff700 (LWP 347103)]
[New Thread 0x7ffff1bfe700 (LWP 347104)]
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
[New Thread 0x7ffbf11ff700 (LWP 347105)]
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
EAL: PCI device 0000:06:00.0 on NUMA socket 0
EAL:   probe driver: 8086:953 spdk_nvme
conn.c: 341:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (143.182.136.79) on iqn.2016-06.io.spdk:disk1 tgt_node1 (143.182.137.32:3260,1), ISID=80f7248c0000, TSIH=1, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:4237:spdk_iscsi_execute: *ERROR*: CmdSN(124749489) error ExpCmdSN=124749425
conn.c:1232:spdk_iscsi_conn_handle_incoming_pdus: *ERROR*: spdk_iscsi_execute() fatal error on iqn.2016-06.io.spdk:disk1,t,0x0001(iqn.2016-06.io.spdk:init,i,0x80f7248c0000)
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
nvme_ctrlr.c: 781:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
conn.c: 341:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (143.182.136.79) on iqn.2016-06.io.spdk:disk1 tgt_node1 (143.182.137.32:3260,1), ISID=80f826d50000, TSIH=2, CID=0, HeaderDigest=off, DataDigest=off
conn.c: 461:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 1): 0

## Possible Solution
No clue

## Steps to Reproduce

Steps are shown in expected and observed behaviors.

## Context (Environment including OS version, SPDK version, etc.)
Fedora 27, master.
**INITIATOR BDEV CONF FILE:**
[iSCSI_Initiator]
        URL iscsi://143.182.137.32/iqn.2016-06.io.spdk:disk1/0 iSCSI0

**TARGET CONF FILE**
[Nvme]
TransportId "trtype:PCIe traddr:0000:06:00.0" Nvme0

[TargetNode1]
  TargetName disk1
  TargetAlias "Data Disk1"
  Mapping PortalGroup1 InitiatorGroup1
  AuthMethod Auto
  AuthGroup AuthGroup1
  UseDigest Auto
  LUN0 Nvme0n1
  QueueDepth 256

[PortalGroup1]
  Portal DA1 143.182.137.32:3260

[InitiatorGroup1]
  InitiatorName ANY
  Netmask 143.182.136.79/24

**Comment by WenqianZong** [comment]
I can reproduce this issue on local server, and find core dump on initator device.
Following is log:

```
TARGET:
[root@waikikibeach43 spdk]# ./app/iscsi_tgt/iscsi_tgt -c ../configure/iscsi.conf.in
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: iscsi --no-shconf -c 0x1 --legacy-mem --file-prefix=spdk_pid49317 ]
EAL: Detected 112 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid49317/mp_socket
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
EAL: PCI device 0000:b4:00.0 on NUMA socket 1
EAL:   probe driver: 8086:953 spdk_nvme
EAL: PCI device 0000:b5:00.0 on NUMA socket 1
EAL:   probe driver: 8086:953 spdk_nvme
EAL: PCI device 0000:d8:00.0 on NUMA socket 1
EAL:   probe driver: 8086:2701 spdk_nvme
EAL: PCI device 0000:d9:00.0 on NUMA socket 1
EAL:   probe driver: 8086:2701 spdk_nvme
conn.c: 341:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (192.168.100.2) on iqn.2016-06.io.spdk:disk1 tgt_node1 (192.168.100.1:3260,1), ISID=804601570000, TSIH=1, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:4237:spdk_iscsi_execute: *ERROR*: CmdSN(1915249317) error ExpCmdSN=1915249253
conn.c:1232:spdk_iscsi_conn_handle_incoming_pdus: *ERROR*: spdk_iscsi_execute() fatal error on iqn.2016-06.io.spdk:disk1,t,0x0001(iqn.2016-06.io.spdk:init,i,0x804601570000)
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c:1093:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor
conn.c: 341:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
nvme_ctrlr.c: 781:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (192.168.100.2) on iqn.2016-06.io.spdk:disk1 tgt_node1 (192.168.100.1:3260,1), ISID=801414e90000, TSIH=2, CID=0, HeaderDigest=off, DataDigest=off
conn.c: 461:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 1): 0
conn.c: 461:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 2): 0
```
```
INITATOR:
[root@waikikibeach41 spdk]# ./test/bdev/bdevperf/bdevperf -c bdev.conf -q 128 -o 4096 -t 30 -w reset
Starting SPDK v18.10-pre / DPDK 18.02.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid2913 ]
EAL: Detected 88 lcore(s)
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Multi-process socket /var/run/.spdk_pid2913_unix
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
Segmentation fault (core dumped)
```

CORE DUMP LOG:
```
[root@waikikibeach41 spdk]# gdb -c /tmp/core-reactor_0-11-0-0-2913-1539221962 test/bdev/bdevperf/bdevperf
GNU gdb (GDB) Fedora 8.0.1-33.fc27
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from test/bdev/bdevperf/bdevperf...done.

warning: core file may not match specified executable file.
[New LWP 2913]
[New LWP 2915]
[New LWP 2914]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `./test/bdev/bdevperf/bdevperf -c bdev.conf -q 128 -o 4096 -t 30 -w reset'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  bdev_iscsi_reset_cb (context=0x227c510, status=251658240, command_data=0x0,
    private_data=0x7f0e55a2a320) at bdev_iscsi.c:326
326             if (tmf_response == ISCSI_TASK_FUNC_RESP_COMPLETE) {
[Current thread is 1 (Thread 0x7f0e5f4852c0 (LWP 2913))]
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.26-21.fc27.x86_64 libaio-0.3.110-9.fc27.x86_64 libibverbs-14-4.fc27.x86_64 libnl3-3.4.0-1.fc27.x86_64 librdmacm-14-4.fc27.x86_64 libuuid-2.30.2-1.fc27.x86_64 numactl-libs-2.0.11-5.fc27.x86_64 openssl-libs-1.1.0g-1.fc27.x86_64 zlib-1.2.11-4.fc27.x86_64
(gdb) bt
#0  bdev_iscsi_reset_cb (context=0x227c510, status=251658240, command_data=0x0,
    private_data=0x7f0e55a2a320) at bdev_iscsi.c:326
#1  0x00007f0e5ee4acef in iscsi_reconnect_cb (iscsi=0x227c510, status=<optimized out>,
    command_data=<optimized out>, private_data=<optimized out>) at connect.c:331
#2  0x00007f0e5ee4a957 in iscsi_login_cb (iscsi=0x227c510, status=<optimized out>,
    command_data=<optimized out>, private_data=0x227eec0) at connect.c:182
#3  0x00007f0e5ee4e80c in iscsi_process_login_reply (iscsi=iscsi@entry=0x227c510,
    pdu=pdu@entry=0x22972f0, in=in@entry=0x2285a40) at login.c:1275
#4  0x00007f0e5ee500b6 in iscsi_process_pdu (iscsi=iscsi@entry=0x227c510, in=in@entry=0x2285a40)
    at pdu.c:548
#5  0x00007f0e5ee5b620 in iscsi_read_from_socket (iscsi=<optimized out>) at socket.c:692
#6  iscsi_tcp_service (iscsi=0x227c510, revents=1) at socket.c:978
#7  0x0000000000437bf0 in bdev_iscsi_poll_lun (lun=<optimized out>) at bdev_iscsi.c:372
#8  0x00000000004433b1 in _spdk_reactor_run (arg=0x1e10280) at reactor.c:514
#9  0x00000000004439a8 in spdk_reactors_start () at reactor.c:677
#10 0x00000000004423f8 in spdk_app_start (opts=opts@entry=0x7ffe86605130,
    start_fn=start_fn@entry=0x407190 <bdevperf_run>, arg1=arg1@entry=0x0, arg2=arg2@entry=0x0)
    at app.c:650
#11 0x000000000040553c in main (argc=<optimized out>, argv=<optimized out>) at bdevperf.c:1013
```

**Comment by peluse** [comment]
Great to know someone else sees this as well!  I may pick this up later but for now its open to anyone who wants to dive in, I just wanted to get it in the system as I discovered it while fixing another issue; have a few other priorities  ahead of this...

**Comment by peluse** [comment]
I don't plan on picking this up but I can if nobody else does.  Not sure what the larger community thinks of how important it is, seems like it would be but then again nobody has reported it before :)  Good one to cover in the Wed eve bug scrub for sure

**Comment by wanqunintel** [comment]
@peluse , Gang will look into this issue.

**Comment by Comphix** [comment]
Looking at this issue.

**Comment by Comphix** [comment]
Tried the latest code, and seems that no core dump hit by so far. For the high queue like 256, it does need more time for the bdevperf to complete. Below is some data, IO took long time for 256 queue depth.

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 256 -o 4096 -t 30 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x1 -m 1024 --legacy-mem --file-prefix=spdk_pid4721 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid4721/mp_socket
EAL: Probing VFIO support...
EAL: WARNING: Master core has no memory on local socket!
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :       9.67 IO/s       0.04 MB/s
 =====================================================
 Total               :       9.67 IO/s       0.04 MB/s

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 128 -o 4096 -t 30 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x1 -m 1024 --legacy-mem --file-prefix=spdk_pid5273 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid5273/mp_socket
EAL: Probing VFIO support...
EAL: WARNING: Master core has no memory on local socket!
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :    5664.00 IO/s      22.12 MB/s
 =====================================================
 Total               :    5664.00 IO/s      22.12 MB/s

**Comment by optimistyzy** [comment]
Hi Gang,

Are you sure? For the latest code, our SPDK iSCSi target will use the
nonblack model for writing out the data.

GangCao <notifications@github.com> 于2018年10月18日周四 下午11:17写道：

**Comment by peluse** [comment]
Note that I was not using loopback, I don't know if that makes any difference or not but I suspect it might if this is somehow related to how long it takes.  Can you try running with 2 machines? Also, when you say " it does need more time for the bdevperf to complete" are you changing a setting somewhere, seeing something in a debug log or what drove you to make this statement?  Thanks!

**Comment by Comphix** [comment]
Just have a simple script to run the testing with queue depth 1024/512/256/128/64 and before each run, log the time, below is the result from the two machines:

Mon Oct 22 20:55:00 CST 2018
./bdevperf -c ./bdev.conf -q 1024 -o 4096 -t 30 -w reset -m 0x100000
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :      97.57 IO/s       0.38 MB/s
 
Duration: 55:36 - 55:00 = 36 seconds
 
Mon Oct 22 20:55:36 CST 2018
./bdevperf -c ./bdev.conf -q 512 -o 4096 -t 30 -w reset -m 0x100000
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :      23.87 IO/s       0.09 MB/s

Duration: 57:33 - 55:36 = 1:57 = 117 seconds

Mon Oct 22 20:57:33 CST 2018
./bdevperf -c ./bdev.conf -q 256 -o 4096 -t 30 -w reset -m 0x100000
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :       9.47 IO/s       0.04 MB/s

Duration: 59:15 - 57:33 = 102 seconds

Mon Oct 22 20:59:15 CST 2018
./bdevperf -c ./bdev.conf -q 128 -o 4096 -t 30 -w reset -m 0x100000
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :    5891.20 IO/s      23.01 MB/s

Duration: 21:00:47 - 20:59:15 = 92 seconds

Mon Oct 22 21:00:47 CST 2018
./bdevperf -c ./bdev.conf -q 64 -o 4096 -t 30 -w reset -m 0x100000
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :   12684.50 IO/s      49.55 MB/s

Observations:
1. When queue depth with 256 and above, the IOPS dropped extremely
2. With larger queue depth, it takes long time for all I/Os completed, the duration is long
3. I also noticed the below error and it looks like not big problem to hang the testing
(conn.c:1132:spdk_iscsi_conn_flush_pdus_internal: *ERROR*: spdk_sock_writev() failed, errno 9: Bad file descriptor)

@xuhuagen, have you waited enough time for the bdevperf to exit? Thanks.

**Comment by peluse** [comment]
thanks!  Yeah #3 is why I submitted the issue. Just a shot in the dark but I wonder if this is related to issue #457?

**Comment by shuhei-matsumoto** [comment]
@peluse  #457 doesn't look related.

It looks that the target received an out-of-order command and exit the current connection correctly. (#457 caused segmentation fault in the target.)

Which of the initiator or the target should be addressed here?
It looks that the behavior of the target looks normal and the behavior of the initiator is against our expectation?
Out-of-order command should not be sent from the initiator to the target?
Out-of-order command is acceptable but the hung in the initiator should not occur?

Clarifying them may be helpful.

**Comment by Comphix** [comment]
@shuhei-matsumoto, so far it seems the initiator needs more time to exit although we set the running time to 30 seconds. I will keep a look. Thanks for the comments.

**Comment by peluse** [comment]
thanks guys! I don't have much insight into this I just filled it as "unexpected" which seems to be the case.  I can repro or test or whatever but other than that am along just for the learning :)

**Comment by Comphix** [comment]
The lower IOPS with high queue depth (>128) is due to below error and trigger the backend reset

(conn.c:1132:spdk_iscsi_conn_flush_pdus_internal: ERROR: spdk_sock_writev() failed, errno 9: Bad file descriptor)

Looking at the implementation, seems like the sock is NULL when hitting this error (9: Bad file descriptor)

**Comment by Comphix** [comment]
For the reset, it is from the error and close the connection as following:
	spdk_iscsi_conn_destruct
		spdk_iscsi_conn_cleanup_backend	
			spdk_iscsi_tgt_node_cleanup_luns
				spdk_scsi_dev_queue_mgmt_task(target->dev, &task->scsi, SPDK_SCSI_TASK_FUNC_LUN_RESET);

**Comment by Comphix** [comment]
So that this behavior looks like expected. Met some out-of-order error, the connection close, reset triggered, IOPS dropped. One thing may need further investigation is why the out-of-order error hit with high queue depth？ Is this also expected? Could be relating to some of the node setting where the initial queue depth has been configured to 256.

**Comment by Comphix** [comment]
Looking at the libiscsi code https://github.com/sahlberg/libiscsi/blob/36548b46692367d2372eb873195b0f47c03daa0e/lib/iscsi-command.c where the iscsi_scsi_command_async() function is called. The IOs are sent in the async mode.

**Comment by Comphix** [comment]
Raised a patch at https://review.gerrithub.io/#/c/spdk/spdk/+/430879/ 

With same configuration as logged in this issue, I can run 512 queue depth without error:

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 512 -o 4096 -t 10 -w write -m 0x100000
Starting SPDK v18.10-pre / DPDK 18.05.0 initialization...
[ DPDK EAL parameters: bdevtest --no-shconf -c 0x100000 -m 1024 --legacy-mem --file-prefix=spdk_pid35773 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/spdk_pid35773/mp_socket
EAL: Probing VFIO support...
EAL: WARNING: Master core has no memory on local socket!
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 20 on socket 0
Running I/O for 10 seconds...
 Logical core: 20
 iSCSI0              :   28328.40 IO/s     110.66 MB/s

**Comment by Comphix** [comment]
@xuhuagen , could you help also to verify this patch and same configuration as Paul logged. Thanks.

**Comment by Comphix** [comment]
Thanks for the update. I've just rebased to the latest code to have the testing again. Previously there is some commits behind.

**Comment by Comphix** [comment]
I do notice with latest code, it takes more time to start the testing "to print out Running I/O for 10 seconds..."

Tested on two machines and looks like it's ok with the patch.

QD: 256, 10s, reset

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 256 -o 4096 -t 10 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: bdevperf --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid91254 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 10 seconds...
 Logical core: 0
 iSCSI0              :   23995.80 IO/s      93.73 MB/s

QD: 512; 10s; reset

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 512 -o 4096 -t 10 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: bdevperf --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid91411 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 10 seconds...
 Logical core: 0
 iSCSI0              :   17839.10 IO/s      69.68 MB/s

Both initiator and target are same commit ID:

[root@node4 bdevperf]# git log
commit 40da9ab5e360d84cd5bb2b705a72eaeb066628cc
Author: Patrick Ohly <patrick.ohly@intel.com>
Date:   Wed Oct 24 09:46:12 2018 +0200

    bdev/rbd: fix error handling after failed rados_connect

As mentioned error, if the target side met the error of CmdSN/ExpCmdSN, the connection will be closed the the controller with be reset, it is the expected behavior. Please confirm:

1. When target logs "resetting controller", at the initiator side, wait enough time to check whether bdevperf can exit by itself.
2. Will it also relate to the network environment although I am also using two machines to cause some out-of-order I/O issue

For the below issue, seems like another one to abort one outstanding request.

nvme_pcie.c:1303:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:185 nsid:1 lba:3690696 len:8

**Comment by Comphix** [comment]
30s testing

QD:512; 30s; reset

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 512 -o 4096 -t 30 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: bdevperf --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid91810 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :    5350.37 IO/s      20.90 MB/s

QD:256; 30s; reset

[root@node4 bdevperf]# ./bdevperf -c ./bdev.conf -q 256 -o 4096 -t 30 -w reset -m 0x100000
Ignoring -m option. Verify can only run with a single core.
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: bdevperf --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid92469 ]
EAL: Detected 32 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 724:spdk_reactors_init: *NOTICE*: Event_mempool creation failed on preferred socket 0.
RING: Cannot reserve memory
reactor.c: 602:spdk_reactor_construct: *NOTICE*: Ring creation failed on preferred socket 0. Try other sockets.
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :   24109.80 IO/s      94.18 MB/s

At target side, it reports NOP timed out so that it seems there is pre-setup not achieved before starting the testing (wait some time to show below log message):
Running I/O for 30 seconds...  <<<<

conn.c:1066:spdk_iscsi_conn_handle_nop: *ERROR*: Timed out waiting for NOP-Out response from initiator
conn.c:1067:spdk_iscsi_conn_handle_nop: *ERROR*:   tsc=0x13a065e6da911c, last_nopin=0x13a046a7e93c98
conn.c:1069:spdk_iscsi_conn_handle_nop: *ERROR*:   initiator=iqn.2016-06.io.spdk:init, target=disk1
conn.c: 460:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 9): 0
conn.c: 340:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (10.67.110.156) on iqn.2016-06.io.spdk:disk1 tgt_node1 (10.67.110.158:3260,1), ISID=804306bc0000, TSIH=10, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 789:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
conn.c: 460:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 10): 0
conn.c: 340:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (10.67.110.156) on iqn.2016-06.io.spdk:disk1 tgt_node1 (10.67.110.158:3260,1), ISID=804262e70000, TSIH=11, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET   <<<<< this is reset from the testing
nvme_ctrlr.c: 789:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 789:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
conn.c: 460:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 11): 0

**Comment by Comphix** [comment]
The timeout of NOP is relating to the host, I've rebooted the host and this issue is not seen at the higher queue depth 256/512

**Comment by peluse** [comment]
let me know if you'd like me to retest w/any particular patch, be happy to

**Comment by Comphix** [comment]
Hi Paul, you may have a try with patch https://review.gerrithub.io/#/c/spdk/spdk/+/430879/ on your original configuration. One noticeable phenomenon is that with larger queue depth like 256, it needs a little long time for the perf to exit although we set the 30 seconds running time. Thanks. Gang

**Comment by peluse** [comment]
ran patch on both init and tgt sides, looks good:
INIT:
sudo ./bdevperf -c ./bdev.conf -q 128 -o 4096 -t 30 -w reset
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: bdevperf --no-shconf -c 0x1 -m 1024 --file-prefix=spdk_pid46835 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
Running I/O for 30 seconds...
 Logical core: 0
 iSCSI0              :   22365.47 IO/s      87.37 MB/s
 =====================================================
 Total               :   22365.47 IO/s      87.37 MB/s

TGT:
sudo ./iscsi_tgt -c x.conf
[sudo] password for peluse:
Starting SPDK v18.10-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: iscsi --no-shconf -c 0x1 --file-prefix=spdk_pid1360832 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
app.c: 602:spdk_app_start: *NOTICE*: Total cores available: 1
reactor.c: 703:spdk_reactors_init: *NOTICE*: Occupied cpu socket mask is 0x1
reactor.c: 490:_spdk_reactor_run: *NOTICE*: Reactor started on core 0 on socket 0
EAL: PCI device 0000:06:00.0 on NUMA socket 0
EAL:   probe driver: 8086:953 spdk_nvme
conn.c: 340:spdk_iscsi_conn_construct: *NOTICE*: Launching connection on acceptor thread
iscsi.c:1934:spdk_iscsi_op_login_notify_session_info: *NOTICE*: Login from iqn.2016-06.io.spdk:init (143.182.136.79) on iqn.2016-06.io.spdk:disk1 tgt_node1 (143.182.137.32:3260,1), ISID=801fdcc30000, TSIH=1, CID=0, HeaderDigest=off, DataDigest=off
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 789:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
iscsi.c:3200:spdk_iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_ctrlr.c: 789:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
conn.c: 460:spdk_iscsi_remove_conn: *NOTICE*: Terminating connections(tsih 1): 0
^C^C[peluse@pels2-64 iscsi_tgt]$

**Comment by liang2yan** [comment]
Since @Comphix 's patch can fix this issue, let's wait for his patch merged.

**Comment by optimistyzy** [comment]
@peluse @Comphix  Can we let someone use this bdevperf with (SPDK iscsi initiator) to test the kernel iSCSI tgt, see what happened? This will be helpful to reduce the issue in iSCSI initiator side ,or the target side. For SPDK iSCSI target side, currently there are some work for the reset related work.

**Comment by jnguo** [comment]
I have used `iscsiadm` to import a SPDK iSCSI target (based on a NVME SSD). And it worked well, without the problem about iodepth too large. At the same time, I/O performance in the initiator side was better than importing the same target device by SPDK initiator.

I guess it's the issue in the SPDK iSCSI initiator side.

**Comment by darsto** [comment]
Bug scrub: @peluse to retest quickly

**Comment by peluse** [comment]
@gongchuang is working on this, see patch in GH for updates.

**Comment by darsto** [comment]
Bug scrub: @peluse please give an update

**Comment by peluse** [comment]
I can't find it. based on the age of this I suggest we close it and if its a real problem it will get reported again.

**Comment by peluse** [comment]
just retested on master, output looks as expected:
nvme_ctrlr.c: 846:spdk_nvme_ctrlr_reset: *NOTICE*: resetting controller
iscsi.c:3492:iscsi_op_task: *NOTICE*: LOGICAL_UNIT_RESET
nvme_pcie.c:1354:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:155 nsid:1 lba:53398512 len:8
nvme_qpair.c: 306:nvme_qpair_print_completion: *NOTICE*: ABORTED - BY REQUEST (00/07) sqid:1 cid:155 cdw0:0 sqhd:0000 p:0 m:0 dnr:1
nvme_pcie.c:1354:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:150 nsid:1 lba:53398520 len:8
nvme_qpair.c: 306:nvme_qpair_print_completion: *NOTICE*: ABORTED - BY REQUEST (00/07) sqid:1 cid:150 cdw0:0 sqhd:0000 p:0 m:0 dnr:1
nvme_pcie.c:1354:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:173 nsid:1 lba:53398784 len:8
nvme_qpair.c: 306:nvme_qpair_print_completion: *NOTICE*: ABORTED - BY REQUEST (00/07) sqid:1 cid:173 cdw0:0 sqhd:0000 p:0 m:0 dnr:1
nvme_pcie.c:1354:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:172 nsid:1 lba:53398792 len:8
nvme_qpair.c: 306:nvme_qpair_print_completion: *NOTICE*: ABORTED - BY REQUEST (00/07) sqid:1 cid:172 cdw0:0 sqhd:0000 p:0 m:0 dnr:1
nvme_pcie.c:1354:nvme_pcie_qpair_abort_trackers: *ERROR*: aborting outstanding command
nvme_qpair.c: 137:nvme_io_qpair_print_command: *NOTICE*: READ sqid:1 cid:168 nsid:1 lba:53398800 len:8
nvme_qpair.c: 306:nvme_qpair_print_completion: *NOTICE*: ABORTED - BY REQUEST (00/07) sqid:1 cid:168 cdw0:0 sqhd:0000 p:0 m:0 dnr:1
